tests/bsim/bluetooth/audio/src/cap_commander_test.c

```diff
@@ -101,6 +101,16 @@ static void cap_discovery_complete_cb(struct bt_conn *conn, int err,
 }
 
 #if defined(CONFIG_BT_VCP_VOL_CTLR)
+static void cap_volume_changed_fail_cb(struct bt_conn *conn, int err)
+{
+	if (err != -ECANCELED) {
+		FAIL("CAP command not cancelled for conn %p: %d\n", conn, err);
+		return;
+	}
+
+	SET_FLAG(flag_volume_changed);
+}
```

Comment:
Instead of a new callback, how about we add a new flag, `flag_cancelled`, and then in `cap_volume_changed_cb` we set the `flag_cancelled` if `err == -ECANCELED`? 

That way we can apply this in a generic way for the other procedures as well

Reply:
That would work as well. The drawback is that with the current implementation a test will fail explicitly with an error message from the callback function. 

If we remove the `FAIL` in the callback, and use the `WAIT_FOR_FLAG` to catch an error we do not get this message anymore. We can work around this (for example using `printk` instead of `FAIL`), but I prefer keeping the current implementation

Reply:
> If we remove the FAIL in the callback, and use the WAIT_FOR_FLAG to catch an error we do not get this message anymore. We can work around this (for example using printk instead of FAIL), but I prefer keeping the current implementation

It's fine that we just printk instead to support this, as the other test will, on error, just time out. 

I think implementing another callback that you have to assign and unassign is going to be more difficult to maintain in the future, so I suggest to stick to a single callback